src-tauri/src/settings.rs

```diff
@@ -29,13 +29,26 @@ impl Default for Tasks {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Agents {
+    pub execution_steps_limit: i64,
+}
```

Comment:
To correctly work with cases where I do already have `Agents` in my config, but didn't have `execution_steps_limit` set:

```suggestion
pub struct Agents {
    #[serde(default = "default_execution_steps_limit")]
    pub execution_steps_limit: i64,
}

fn default_execution_steps_limit() -> i64 {
    // It's better to make this a constant, btw
    12
}
```